### PR TITLE
Have findbugs run as part of the build

### PR DESCRIPTION
**Pull request by big-andy-coates**
### Description 
...  and fail the build if any issues are found.

Initially, its only checking with a high threshold, (i.e. really bad things), but we can bring this down over time.

I've also moved us over to `hamcrest-all` from `hamcrest-core` as it gives us a wider range of matchers and I've seen issues in the past running both.

### Testing done 
`mvn clean install` + run up server & CLI and fire off some commands.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/TopicStream.java [hunk 1]
```diff
@@ -32,46 +31,52 @@
...
     private static final Logger log = LoggerFactory.getLogger(RecordFormatter.class);
-    private final SchemaRegistryClient schemaRegistryClient;
+
+    private final KafkaAvroDeserializer avroDeserializer;
     private final String topicName;
+    private final DateFormat dateFormat =
+        SimpleDateFormat.getDateTimeInstance(3, 1, Locale.getDefault());
 
-    private Format format = Format.UNDEFINED;
+    private Formatter formatter = null;
```
Comment (dguy) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/TopicStream.java]: nit: null is not required. 4 characters takes up too much room on the screen ;-)

File: ksql-common/pom.xml [hunk 2]
```diff
@@ -56,6 +56,12 @@
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
             <scope>test</scope>
+            <exclusions>
```
Comment (apurvam) [inline ksql-common/pom.xml]: Does this mean that every new module will have to explicitly exclude `hamcrest-core`?

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/TopicStream.java [hunk 3]
```diff
@@ -82,116 +87,144 @@ public RecordFormatter(SchemaRegistryClient schemaRegistryClient, String topicNa
...
+      return formatter == null ? Format.UNDEFINED : formatter.getFormat();
+    }
+
+    private Formatter getFormatter(final ConsumerRecord<String, Bytes> record) {
+      return Arrays.stream(Format.values())
+          .map(f -> f.maybeGetFormatter(topicName, record, avroDeserializer, dateFormat))
+          .filter(Optional::isPresent)
+          .map(Optional::get)
+          .findFirst()
+          .orElseThrow(() -> new RuntimeException("Unexpected"));
```
Comment (apurvam) [inline ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/TopicStream.java]: "Unexpected" seems like a terse message. Maybe "Could not find an appropriate formatter"?

File: ksql-common/pom.xml (hunk 2 above)
Comment (big-andy-coates) [reply ksql-common/pom.xml]: Yes. In favour of hamcrest-all

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/TopicStream.java (hunk 3 above)
Comment (big-andy-coates) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/TopicStream.java]: It should never happen. The code always defaults to String.

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/TopicStream.java [hunk 4]
```diff
@@ -82,116 +87,144 @@ public RecordFormatter(SchemaRegistryClient schemaRegistryClient, String topicNa
...
       @Override
-      public boolean isFormat(
-          String topicName, ConsumerRecord<String, Bytes> record,
-          SchemaRegistryClient schemaRegistryClient
-      ) {
-        this.topicName = topicName;
+      public Optional<Formatter> maybeGetFormatter(
+          final String topicName,
+          final ConsumerRecord<String, Bytes> record,
+          final KafkaAvroDeserializer avroDeserializer,
```
[the remark this answers is not in the document]
Comment (big-andy-coates) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/TopicStream.java]: Really this code shouldn't be using an Enum for the different formatter impls. However, as it does... the enums should have mutable state, which the do in the previous version. Hence I can either pass through the schema registry (and create an Avro deserializer on each invocation), or pass through the avro deserializer. I chose the latter.  When we add more we can restructure this code.

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/TopicStream.java (hunk 1 above)
Comment (big-andy-coates) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/TopicStream.java]: Where you just trying to find something to complain about? ;)

File: ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/TopicStream.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (xvrl) [reply ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/streaming/TopicStream.java]: here's a more generic version of this code that supports both keys/values and converts to a more easily parseable format, we might want to consolidate and use that instead? https://github.com/confluentinc/blueway/blob/master/control-center/src/main/java/io/confluent/controlcenter/rest/res/OmgDeserializer.java

File: pom.xml [hunk 5]
```diff
@@ -344,8 +341,28 @@
                     <findbugsXmlWithMessages>true</findbugsXmlWithMessages>
                     <xmlOutput>true</xmlOutput>
                     <excludeFilterFile>findbugs/findbugs-exclude.xml</excludeFilterFile>
+                    <effort>Max</effort>
```
Comment (xvrl) [inline pom.xml]: findbugs plugin versions and some of those configs are already defined as part of the common pom. You should probably inherit those and only add the settings that are not already in common (e.g. fail-on-error). This also ensure we don't have inconsistent plugin versions across projects.

**Pull request merged**